### ISSUE TITLE
"Undo" and "Save" commands don't work; changes are instantly written to file

### ISSUE THREAD

**Issue by skycommand** [Zettlr/Zettlr#1756]
## Description
Zettlr can cause irreversible damage to Markdown files because neither the "Edit -> Undo" command, not the "FIle -> Save" commands do not work. Their shortcuts (Ctrl+Z and Ctrl+S) work, but in the case of Ctrl+S, it is completely pointless.

## Reproducing
1. Use a text editor other than Zettlr (e.g. Notepad, Visual Studio Code, or Notepad++) to create a simple Markdown document as follows. Save it as `Test.markdown`.
    ~~~markdown
    # Test
    
    | Test | Test | Test | Test |
    | ---- | ---- | ---- | ---- |
    | 1    | 1    | 1    | 1    |
    | 2    | 2    | 2    | 2    |
    | 3    | 3    | 3    | 3    |
    | 4    | 4    | 4    | 4    |
    ~~~
2. Open `Test.markdown` in Zettlr.
3. Make a random change to the file, e.g, delete one of the table columns.
4. Try to undo the change by opening the "Edit" menu and selecting "Undo".
5. Open `Test.mardown` in Notepad. (If Notepad is already open with this file loaded, open a new instance. Feel free to either close or ignore the existing instance.)
6. Try to undo the change by pressing the Ctrl+Z key combination on the keyboard
7. Open the "File" menu and select "Save". Pay attention to the change indicator (the asterisk icon) on the document tab.
8. Press Ctrl+S key combination on the keyboard. Pay attention to the change indicator (the asterisk icon) on the document tab.

## Expected behavior

- Steps 4 and 6 must have the same results: The change must get undone. 
    - Actual behavior: In step 4, nothing happens.
- Steps 7 and 8 must have the same results: The changes to the file must get saved.
    - Actual behavior: In step 7 nothing happens. In step 8, Zettlr gives the impression of saving changes, i.e. the change indicator disappears. In reality, the changes have already been saved regardless of this command.
- In step 5, the copy `Test.markdown` that opens in Notepad must not contain your unsaved changes because you have not issued a "Save" command yet. The change indicator also says that the changes are unsaved.
    - Actual behavior: Your changes have already been saved to the file.
- A co-worker asked me if the menus are working at all. I checked:
    - The "Window -> Close" menu command works.
    - The "Edit -> Paste" command does not work.
    - The "Edit -> Paste without style" command works.

## Platform
 - OS and version: Windows 10 version 20H2 x64
 - Zettlr Version: 1.8.7

## Additional information

- I tried searching the Preferences and Zettlr documentations for "autosave". I found nothing.

**Comment by nathanlesage** [comment]
Interesting … thank you for checking that so thoroughly. I will pay extra attention while moving to Zettlr 2.0. I actually didn't know that the menu shortcuts stopped working at some point due to me mainly using keystrokes for that.

(Interestingly, though, this has to do with the custom menus I introduced. I have to admit, it was a mistake and I'll be reverting that on the next iteration).

With regard to save: It was a common wish for users to have documents save automatically, but yes, this basically nukes the undo-history. I'll revert back to manual save for 2.0, as I feel Zettlr is becoming more serious and hence auto-save doesn't apply (for instance for my very own workflows).

**Comment by waterfallpit** [comment]
I am testing saving effects and I only see changes in the file after file is auto-saved by Zettlr or I press Ctrl-S. I am using Notepad++ with "Reload from file" function for comparison. Win 10 and 1.8.7 here as well.
"Edit -> Undo", however, does not work, as you described, but "File -> Save" does work for me.

**Comment by nathanlesage** [comment]
The menu items somehow work again. Please notify me if they refuse at some point again.

**Comment by skycommand** [comment]
Exciting. I will test them as soon as I reach home. That means Monday.

**Comment by skycommand** [comment]
@nathanlesage As you asked me, I downloaded the latest version of Zettlr and tested it. I'm afraid this bug hasn't been fixed. In fact, it has gotten worse. Even Ctrl+Z didn't work this time.

**Comment by skycommand** [comment]
I've recorded a video that shows the problem in version 1.8.9:

https://user-images.githubusercontent.com/17097175/121824299-667a2480-ccc0-11eb-9011-d15c2edb72cc.mp4

**Comment by waterfallpit** [comment]
Move the  cursor outside of the table first and then try Ctrl-Z. It works for me that way. Edit->Undo/Redo don't work though (1.8.9 on Win 10).

**Comment by nathanlesage** [comment]
> As you asked me, I downloaded the latest version of Zettlr and tested it.

Nah, the fix is in 2.0-dev, not the latest released one. Please either wait for the betas to start, and then test it again, or you can also build it locally and test it out right away. The instructions to that are in the README, but you need to have a little bit of knowledge with the command line.
